tools/node-lint-md-cli-rollup/src/list-released-versions-from-changelogs.mjs

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env node
+
+import fs from 'node:fs';
+import { createInterface } from 'node:readline';
+
+const dataFolder = new URL('../../../doc/changelogs/', import.meta.url);
+
+const result = [];
+async function getVersionsFromFile(file) {
+  const input = fs.createReadStream(file);
+  let toc = false;
+  for await (const line of createInterface({
+    input,
+    crlfDelay: Infinity,
+  })) {
+    if (toc === false && line === '<table>') {
+      toc = true;
+    } else if (toc && line[0] !== '<') {
+      input.close();
+      return;
+    } else if (toc && line.startsWith('<a')) {
+      result.push(line.slice(line.indexOf('>') + 1, -'</a><br/>'.length));
+    }
+  }
+}
+
+const filesToCheck = [];
+
+const dir = await fs.promises.opendir(dataFolder);
+for await (const dirent of dir) {
+  if (dirent.name.endsWith('.md')) {
```

Comment:
```suggestion
  if (/CHANGELOG_.*\.md/.test(dirent.name)) {
```
Since we're validating the filenames before passing them to `getVersionsFromFile
`,
should we use this check instead?

Reply:
Yes, the check was there only to filter `.` and `..`, I guess this would be simpler and more explicit:
```suggestion
  if (dirent.isFile()) {
```